src/codegen/constant_folding.rs

```diff
@@ -386,1192 +387,192 @@ fn expression(
             overflowing,
             left,
             right,
-        } => {
-            let left = expression(left, vars, cfg, ns);
-            let right = expression(right, vars, cfg, ns);
-
-            if let (
-                Expression::NumberLiteral { value: left, .. },
-                Expression::NumberLiteral { value: right, .. },
-            ) = (&left.0, &right.0)
-            {
-                bigint_to_expression(loc, ty, left.add(right))
-            } else {
-                (
-                    Expression::Add {
-                        loc: *loc,
-                        ty: ty.clone(),
-                        overflowing: *overflowing,
-                        left: Box::new(left.0),
-                        right: Box::new(right.0),
-                    },
-                    left.1 && right.1,
-                )
-            }
-        }
+        } => add(loc, ty, *overflowing, left, right, vars, cfg, ns),
```

Comment:
Did you make any changes apart from factoring code into functions?

Reply:
I united the calculation of hashes in a single function. The match statement had different branches for each of them and I found that to be repeated code. Please, see `fn builtin_hash` on line 1279.